### PR TITLE
Add documentation about security advisories

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

**Review by sigurdm — APPROVED** [review]

File: src/tools/pub/security-advisories.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+title: Security advisories
+description: Use security advisories to inform and be informed about security vulnerabilities.
+---
+
+Security advisories are a means to report information about security
+vulnerabilities. Pub uses the [Github Advisory
+Database][https://github.com/advisories] for publishing security advisories for
+Dart and Flutter packages. 
+
+To create an advisory in your Github repo use Github's security advisory
```
Comment (jonasfj) [inline src/tools/pub/security-advisories.md:11]:
```suggestion
To create an advisory in your Github repository use Github's security advisory
```

File: src/tools/pub/security-advisories.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  [^0]: https://github.com/advisories/GHSA-4rgh-jx4f-qfcq
+```
+
+In such a case we recommend you follow the link and review the advisory. After
+reviewing, if you asses that this vulnerability is affecting your package, you
+should strongly consider upgrading to another version of your dependency.
+
+
+### Ignoring security advisories
+If a security advisory is not relevant for your package, you can suppress the
```
Comment (jonasfj) [inline src/tools/pub/security-advisories.md:37]:
```suggestion
If a security advisory is not relevant for your application, you can suppress the
```

**Review by jonasfj — APPROVED** [review]

File: src/tools/pub/security-advisories.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+---
+title: Security advisories
+description: Use security advisories to inform and be informed about security vulnerabilities.
+---
+
+Security advisories are a means to report information about security
+vulnerabilities. Pub uses the [Github Advisory
+Database][https://github.com/advisories] for publishing security advisories for
+Dart and Flutter packages.
```
Comment (MaryaBelanger) [inline src/tools/pub/security-advisories.md:9]:
```suggestion
vulnerabilities. Pub uses the [Github Advisory Database][]
for publishing security advisories for Dart and Flutter packages. 
```

added the link to the bottom

File: src/tools/pub/security-advisories.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+```
+name: myapp
+dependencies:
+  foo: ^1.0.0
+ignored_advisories:
+ - GHSA-4rgh-jx4f-qfcq
+```
+
+The list of `ignored_advisories` only affects the root package. I.e, ignored
+advisories in your dependencies will have no effect on your resolution.
```
Comment (MaryaBelanger) [inline src/tools/pub/security-advisories.md:50]:
```suggestion
advisories in your dependencies will have no effect on your resolution.

[Github Advisory Database]: [https://github.com/advisories]
[here]: https://docs.github.com/en/code-security/security-advisories/working-with-repository-security-advisories/creating-a-repository-security-advisory

```

File: src/tools/pub/security-advisories.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+---
+
+Security advisories are a means to report information about security
+vulnerabilities. Pub uses the [Github Advisory
+Database][https://github.com/advisories] for publishing security advisories for
+Dart and Flutter packages. 
+
+To create an advisory in your Github repository use Github's security advisory
+reporting mechanism as explained
+[here][https://docs.github.com/en/code-security/security-advisories/working-with-repository-security-advisories/creating-a-repository-security-advisory].
```
Comment (MaryaBelanger) [inline src/tools/pub/security-advisories.md:13]:
```suggestion
To create an advisory in your Github repository, use Github's security advisory
reporting mechanism as explained [here][].
```

added link to bottom

File: src/tools/pub/security-advisories.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+First you create a draft security advisory, which will then be reviewed by
+Github and ingested into the central database.
+
+
+## Security advisories in the pub client
```
Comment (MaryaBelanger) [inline src/tools/pub/security-advisories.md:18]:
```suggestion
## Security advisories in the pub client

```

File: src/tools/pub/security-advisories.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+The pub client surfaces security advisories at dependency resolution. For
+instance when running `dart pub get` you will get the following output:
```
Comment (MaryaBelanger) [inline src/tools/pub/security-advisories.md:20]:
```suggestion
The pub client surfaces security advisories at dependency resolution.
For instance, when running `dart pub get` you will get the following output:
```

File: src/tools/pub/security-advisories.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+Dependencies are affected by security advisories:
+  [^0]: https://github.com/advisories/GHSA-4rgh-jx4f-qfcq
+```
+
+In such a case we recommend you follow the link and review the advisory. After
+reviewing, if you asses that this vulnerability is affecting your package, you
+should strongly consider upgrading to another version of your dependency.
+
+
+### Ignoring security advisories
```
Comment (MaryaBelanger) [inline src/tools/pub/security-advisories.md:36]:
```suggestion
### Ignoring security advisories

```

File: src/tools/pub/security-advisories.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+If a security advisory is not relevant for your application, you can suppress the
+warning by adding the advisory to the list of `ignored_advisories` in the
+`pubspec.yaml` of your package. For example:
+
+```
```
Comment (MaryaBelanger) [inline src/tools/pub/security-advisories.md:41]: If you want yaml syntax highlighting you can do this:
````suggestion
```yaml
````

File: src/tools/pub/security-advisories.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+name: myapp
+dependencies:
+  foo: ^1.0.0
+ignored_advisories:
+ - GHSA-4rgh-jx4f-qfcq
+```
+
+The list of `ignored_advisories` only affects the root package. I.e, ignored
```
Comment (MaryaBelanger) [inline src/tools/pub/security-advisories.md:49]:
```suggestion
The list of `ignored_advisories` only affects the root package. Ignored
```

**Review by MaryaBelanger — COMMENTED** [review]
LGTM, but some broken markdown so I left suggestions

File: src/tools/pub/security-advisories.md (hunk 4 above)
Comment (szakarias) [reply src/tools/pub/security-advisories.md:50]: Do we need the brackets on line 52?

Comment (MaryaBelanger) [reply src/tools/pub/security-advisories.md:50]: Oh, my mistake. No the brackets aren't needed after the colon. I'll fix it!

File: src/tools/pub/security-advisories.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+dependencies:
+  foo: ^1.0.0
+ignored_advisories:
+ - GHSA-4rgh-jx4f-qfcq
+```
+
+The list of `ignored_advisories` only affects the root package. Ignored
+advisories in your dependencies will have no effect on your resolution.
+
+[Github Advisory Database]: [https://github.com/advisories]
```
Comment (MaryaBelanger) [inline src/tools/pub/security-advisories.md:52]:
```suggestion
[Github Advisory Database]: https://github.com/advisories
```

**Review by MaryaBelanger — APPROVED** [review]
Thanks!

**Comment by parlough** [comment]
Exciting stuff. Thanks all!

**Pull request merged**